### PR TITLE
[PM-11904] - send options form

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11904

## 📔 Objective

This PR adds the new send options form.

## 📸 Screenshots

Screenshot 2024-09-18 at 5 15 03 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/tools/send/send-ui/src/send-form/components/send-details/send-text-details.component.ts
Comment (jaasen-livefront) [inline libs/tools/send/send-ui/src/send-form/components/send-details/send-text-details.component.ts:1]: removed unnecessary code

File: libs/tools/send/send-ui/src/send-form/components/send-details/send-file-details.component.ts
Comment (jaasen-livefront) [inline libs/tools/send/send-ui/src/send-form/components/send-details/send-file-details.component.ts:1]: removed unnecessary code

File: libs/tools/send/send-ui/src/send-form/components/options/send-options.component.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  IconButtonModule,
+  SectionComponent,
+  SectionHeaderComponent,
+  TypographyModule,
+} from "@bitwarden/components";
+
+import { SendFormConfig } from "../../abstractions/send-form-config.service";
+import { SendFormContainer } from "../../send-form-container";
+
+export type SendOptionsForm = FormGroup<{
```
Comment (djsmith85) [inline libs/tools/send/send-ui/src/send-form/components/options/send-options.component.ts:21]: ⚠️ Creating and exporting a type that would need to be extended anytime we want to add a field and also extending the formGroup itself, seems unnecessary. The consuming part of the exported type, also doesn't care about the fields included.

You should be able to model it, the way `sendDetailsForm` is registered within the `send-form-container`: https://github.com/bitwarden/clients/pull/11142/files#diff-f01d4154fd872bd55c155a5a4d40e6ca737c00347212adb4c2dc69d561ec4004R13

This also applies to the send-file-details and send-text-details components.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: libs/tools/send/send-ui/src/send-form/components/options/send-options.component.ts (hunk 1 above)
Comment (jaasen-livefront) [reply libs/tools/send/send-ui/src/send-form/components/options/send-options.component.ts:21]: @djsmith85 Agreed this is unnecessary. I've fixed the options form as well as the others.

File: apps/browser/src/_locales/en/messages.json [hunk 2]
```diff
@@ -2267,6 +2270,23 @@
...
+  "limitSendViewsHintWithCount": {
+    "message": "No one can view this Send after the limit is reached. $ACCESSCOUNT$ views left",
+    "description": "Displayed under the limit views field on Send",
+    "placeholders": {
+      "accessCount": {
+        "content": "$1",
+        "example": "2"
+      }
+    }
+  },
```
[the remark this answers is not in the document]
Comment (jaasen-livefront) [reply apps/browser/src/_locales/en/messages.json]: Agreed!

File: apps/browser/src/_locales/en/messages.json [hunk 3]
```diff
@@ -2409,6 +2429,10 @@
     "message": "Optionally require a password for users to access this Send.",
     "description": "'Send' is a noun and the name of a feature called 'Bitwarden Send'. It should not be translated."
   },
+  "sendPasswordDescV2": {
+    "message": "Require this password to view the Send.",
+    "description": "'Send' is a noun and the name of a feature called 'Bitwarden Send'. It should not be translated."
+  },
```
[the remark this answers is not in the document]
Comment (jaasen-livefront) [reply apps/browser/src/_locales/en/messages.json:2435]: You're absolutely right. Thanks for finding this!

**Review by djsmith85 — APPROVED** [review]
Nice work!

**Pull request merged**